database/rrdhost.c

```diff
@@ -206,6 +200,115 @@ void set_host_properties(RRDHOST *host, int update_every, RRD_MEMORY_MODE memory
 // ----------------------------------------------------------------------------
 // RRDHOST - add a host
 
+static void rrdhost_initialize_rrdpush(RRDHOST *host,
+                                       unsigned int rrdpush_enabled,
+                                       char *rrdpush_destination,
+                                       char *rrdpush_api_key,
+                                       char *rrdpush_send_charts_matching
+) {
+    if(rrdhost_flag_check(host, RRDHOST_FLAG_INITIALIZED_RRDPUSH)) return;
+    rrdhost_flag_set(host, RRDHOST_FLAG_INITIALIZED_RRDPUSH);
+
+    sender_init(host);
+    netdata_mutex_init(&host->receiver_lock);
+
+    host->rrdpush_send_enabled     = (rrdpush_enabled && rrdpush_destination && *rrdpush_destination && rrdpush_api_key && *rrdpush_api_key) ? 1 : 0;
+    host->rrdpush_send_destination = (host->rrdpush_send_enabled)?strdupz(rrdpush_destination):NULL;
+
+    if (host->rrdpush_send_destination)
+        host->destinations = destinations_init(host->rrdpush_send_destination);
+
+    host->rrdpush_send_api_key     = (host->rrdpush_send_enabled)?strdupz(rrdpush_api_key):NULL;
+    host->rrdpush_send_charts_matching = simple_pattern_create(rrdpush_send_charts_matching, NULL, SIMPLE_PATTERN_EXACT);
+
+    host->rrdpush_sender_pipe[0] = -1;
+    host->rrdpush_sender_pipe[1] = -1;
+    host->rrdpush_sender_socket  = -1;
+
+    //host->stream_version = STREAMING_PROTOCOL_CURRENT_VERSION;        Unused?
+#ifdef ENABLE_HTTPS
+    host->ssl.conn = NULL;
+    host->ssl.flags = NETDATA_SSL_START;
+    host->stream_ssl.conn = NULL;
+    host->stream_ssl.flags = NETDATA_SSL_START;
+#endif
+}
+
+static void rrdhost_initialize_health(RRDHOST *host,
+                                      int is_localhost
+                                      ) {
+    if(!host->health_enabled || rrdhost_flag_check(host, RRDHOST_FLAG_INITIALIZED_HEALTH)) return;
+    rrdhost_flag_set(host, RRDHOST_FLAG_INITIALIZED_HEALTH);
+
+    rrdfamily_index_init(host);
+    rrdcalctemplate_index_init(host);
+    rrdcalc_rrdhost_index_init(host);
+    host->rrdvars = rrdvariables_create();
+
+    host->health_default_warn_repeat_every = config_get_duration(CONFIG_SECTION_HEALTH, "default repeat warning", "never");
```

Comment:
Hello @ktsaou,

I am running different tests on your PR, and I coverity is reporting:

```sh
*** CID 357071:  High impact quality  (Y2K38_SAFETY)
/database/rrdhost.c: 254 in rrdhost_initialize_health()
248         host->health_default_warn_repeat_every = config_get_duration(CONFIG_SECTION_HEALTH, "default repeat warning", "never");
249         host->health_default_crit_repeat_every = config_get_duration(CONFIG_SECTION_HEALTH, "default repeat critical", "never");
250     
251         host->health_log.next_log_id = 1;
252         host->health_log.next_alarm_id = 1;
253         host->health_log.max = 1000;
>>>     CID 357071:  High impact quality  (Y2K38_SAFETY)
>>>     A "time_t" value is stored in an integer with too few bits to accommodate it.  The expression "now_realtime_sec()" is cast to "uint32_t".
254         host->health_log.next_log_id = (uint32_t)now_realtime_sec();
255         host->health_log.next_alarm_id = 0;
256     
257         long n = config_get_number(CONFIG_SECTION_HEALTH, "in memory max health log entries", host->health_log.max);
258         if(n < 10) {
259             error("Host '%s': health configuration has invalid max log entries %ld. Using default %u", rrdhost_hostname(host), n, host->health_log.max);

** CID 329593:    (TAINTED_SCALAR)
```

Reply:
Yes, this is by design.